### DISCUSSION TITLE
Conditional "Run Only if" dont works

### DISCUSSION THREAD

**Discussion by P3df4r** [ToolJet/ToolJet#11164] [Q&A]
Hi everyone, recently i discover a Tooljet (greatfull work) and inicialize a learn how to make my models.

I was defined some conditional to add some informations in my mongodb, but dont worked. I thinking that error is my, but i checked the boolean value and dont limited the request.
I record a video to help and detailed the another test realized.

Steps the second test:
1 - Register a text input denominated 'product_name_register', extracted to 'textinput1' (Denominated 'nome' in video)
2 - Register a value to variable with '{{components.textinput1.value}}'
3 - Set a button to send to mongodb the informations inputed to user. The variables are defined to user press the button end run the query (mongodb) if tha "Run only if" to satisfaction.
4 - I tested with conditions:
4.1- {{variables.product_name_register}} === 1 
4.2 - {{variables.product_name_register}} === '1' 
5 - I tested 3 scenarios:
5.1 - Insert 1
5.2 - Insert '1'
5.3 - Insert 2
6 - In all scenarios the 'Run only if' dont stop the run query (the run query only stop if 5.2 scenario with mongodb, because dont accept the parentheses)

Youtube link the first test: https://youtu.be/UaHPlvcxv7g

Some information: 
Docker: tooljet/try EE-LTS-latest
Mongo version 8.0.3
Mongosh 2.3.2

**Comment by PriteshKiri** [comment]
Hey @P3df4r,

Thanks for sharing the issue in detail.

Can you please try adding the condition {{components.textinput1.value==="1"}} in the  'Run only if'?

Have shared the screenshot for reference.

Please let me know if you are still having issues.

<img width="829" alt="Screenshot 2024-11-06 at 3 14 05 PM" src="https://github.com/user-attachments/assets/cccc1a2a-2c2c-4081-9cd1-0eed0b4507d1">
